src/Enzo/enzo_EnzoCenteredFieldRegistry.hpp

```diff
@@ -169,12 +169,39 @@ class EnzoCenteredFieldRegistry
   /// There are 2 ways that a name can be associated with a quantity
   ///   1. They can exactly match the name of the quantity.
   ///   2. For VECTOR quantities, they can be composed of the quantity name
-  ///      followed by a 2 character suffix. If ijk_suffix is true, then the
-  ///      suffixes are {'_i', '_j', '_k'}. Otherwise they are
-  ///      {'_x', '_y', '_z'}.
+  ///      followed by a 2 character suffix. The `ijk_suffix` argument
+  ///      determines the suffixes, as in ``try_get_vector_component``.
   static std::string get_actively_advected_quantity_name
   (std::string name, bool ijk_suffix) noexcept;
 
+  /// Identifies the vector component (if any) associated with a name.
+  ///
+  /// For all names consisting of 3 or more characters, this checks whether the
+  /// last 2-characters in the string match a list of known suffixes (the
+  /// list's contents depend on the `ijk_suffix` parameter). Upon success, this
+  /// returns the character corresponding to the suffix. In all other cases,
+  /// this returns the null character, `'\0'`.
+  ///
+  /// @param name The name that will be checked
+  /// @param ijk_suffix When true, the suffixes are `{'_i', '_j', '_k'}`, and
+  ///     the function returns `'i'`, `'j'`, or `'k'` upon a successful match.
+  ///     Otherwise, the suffixes are `{'_x', '_y', '_z'}`, and the function
+  ///     returns `'x'`, `'y'`, or `'z'` upon a successful match.
+  static char try_get_vector_component(const std::string& name,
+                                       bool ijk_suffix) noexcept
+  {
+    const std::array<char,3> ref = {ijk_suffix ? 'i' : 'x',
+                                    ijk_suffix ? 'j' : 'y',
+                                    ijk_suffix ? 'k' : 'z'};
+    std::size_t len = name.length();
+    if ((len >= 3) && (name[len - 2] == '_')){
+        if (std::find(ref.begin(), ref.end(), name[len - 1]) != ref.end()){
+            return name[len - 1];
+        }
+    }
+    return '\0';
+  }
```

Comment:
> Is there a way that you could "maliciously" add fields to cause problems? I.e. if I added a field called "planet_x" and looked for the vector component of "planet". This is a contrived example but I could imagine users adding fields with _x and _i subscripts for reasons other than Cartesian coordinates.
> 
> Maybe if one added He_i and He_ii -- wuld these cases cause problems or would trying for vector components just be nonsense?

In principle this is possible. In practice, all 3 locations where this function is being called, we make sure that the prefix is a known vector quantity (like `"velocity"` or `"bfield"`). 

I could consolidate this function with `EnzoCenteredFieldRegistry::get_actively_advected_quantity_name` to prevent all chances of that situation occurring, but it might be better to do that as part of PR #145 (for performance reasons).

But I'll look into that later this evening.

Reply:
It turns out that performance wasn't a concern at all. I made the change. Hopefully this addresses your concerns